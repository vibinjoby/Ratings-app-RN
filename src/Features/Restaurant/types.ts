export interface DetailsRouteProps {
  key: string
  name: string
  params: {
    restaurantId: number
  }
}
