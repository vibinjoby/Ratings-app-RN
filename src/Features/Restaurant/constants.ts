export const ScreenNames = {
  DETAILS: 'Details',
}
