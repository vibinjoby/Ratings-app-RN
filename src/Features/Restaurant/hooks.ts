import { useMutation, useQuery } from '@apollo/client'

import { CreateReviewInput } from '../../../__generated__/globalTypes'
import { CreateReviewMutation } from './gql/CreateReviewMutation'
import { RestaurantDetails } from './gql/RestaurantDetails'
import { SaveReviewReplyMutation } from './gql/SaveReviewReplyMutation'

export const useRestaurantDetails = (id: number) => {
  const { ...rest } = useQuery(RestaurantDetails, {
    variables: {
      id,
    },
    fetchPolicy: 'cache-and-network',
  })
  return { ...rest }
}

export const useAddReview = (createReviewInput: CreateReviewInput) => {
  const [onAddRestaurantMutate, { ...rest }] = useMutation(CreateReviewMutation, {
    variables: {
      createReviewInput,
    },
    fetchPolicy: 'network-only',
  })
  return { onAddRestaurantMutate, ...rest }
}

export const useSaveOwnerReply = () => {
  const [onSaveReply, { ...rest }] = useMutation(SaveReviewReplyMutation, {
    fetchPolicy: 'network-only',
  })
  return { onSaveReply, ...rest }
}
