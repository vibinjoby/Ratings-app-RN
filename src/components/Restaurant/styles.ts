import { StyleSheet } from 'react-native'

import Colors from '../../utilities/colors'
import Typography from '../../utilities/typography'

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  imgBg: {
    borderRadius: 10,
    height: 80,
    width: 80,
    marginEnd: 20,
  },
  starImg: {
    marginRight: 5,
  },
  ratings: {
    ...Typography.Caption1.regular,
    fontSize: 12,
    marginRight: 10,
  },
  ratingWrapper: {
    paddingHorizontal: 5,
    paddingTop: 2,
    backgroundColor: Colors.appOrange,
    flexDirection: 'row',
    borderRadius: 8,
    margin: 7,
  },
  starContainer: {
    flexDirection: 'row',
    marginVertical: 5,
  },
  restaurantTitle: {
    ...Typography.Title1.regular,
    marginTop: 5,
    fontSize: 24,
    maxWidth: 200,
  },
  reviews: {
    ...Typography.Caption1.regular,
    fontSize: 14,
  },
  chevron: {
    position: 'absolute',
    right: 10,
  },
})
