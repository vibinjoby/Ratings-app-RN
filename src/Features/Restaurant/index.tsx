import React from 'react'
import { createStackNavigator } from '@react-navigation/stack'
import { useRoute } from '@react-navigation/native'

import { ScreenNames } from './constants'
import RestaurantDetails from './containers/RestaurantDetails'
import { DetailsRouteProps } from './types'

const RestaurantDetailsNav = () => {
  const route = useRoute<DetailsRouteProps>()
  const Stack = createStackNavigator()

  return (
    <Stack.Navigator screenOptions={{ headerShown: false }}>
      <Stack.Screen
        name={ScreenNames.DETAILS}
        component={RestaurantDetails}
        initialParams={{ restaurantId: route.params.restaurantId }}
      />
    </Stack.Navigator>
  )
}

export default RestaurantDetailsNav
