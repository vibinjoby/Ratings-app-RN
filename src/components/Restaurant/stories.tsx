import React from 'react'
import { View, StyleSheet } from 'react-native'
import { storiesOf } from '@storybook/react-native'

import Restaurant from './component'

storiesOf('Components', module).add('Restaurant', () => (
  <View style={styles.container}>
    <Restaurant
      restaurantName="California Roll"
      ratings={4.5}
      reviewsCount={15}
      onPress={() => console.log('pressed')}
    />
  </View>
))
const styles = StyleSheet.create({
  container: {
    padding: 20,
  },
})
