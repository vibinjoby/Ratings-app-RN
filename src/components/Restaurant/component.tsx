import React from 'react'
import { TouchableOpacity, View, Text, Image } from 'react-native'
import MaterialCommIcons from 'react-native-vector-icons/MaterialCommunityIcons'

import commonConst from '../../configs/commonConst'
import Colors from '../../utilities/colors'
import Stars from '../Stars'
import styles from './styles'

export interface RestaurantProps {
  restaurantName: string
  ratings: number
  reviewsCount: number
  onPress: () => void
}

const Restaurant: React.FC<RestaurantProps> = ({
  restaurantName,
  ratings,
  onPress,
}: RestaurantProps) => {
  return (
    <TouchableOpacity activeOpacity={0.8} onPress={onPress}>
      <View style={styles.container}>
        <Image source={{ uri: commonConst.DUMMY_PIC }} style={styles.imgBg} />
        <View>
          <Text style={styles.restaurantTitle} numberOfLines={1}>
            {restaurantName}
          </Text>
          <View style={styles.starContainer}>
            <Text style={styles.ratings}>{ratings}</Text>
            <Stars
              selectedColor={Colors.appOrange}
              selectable={false}
              selectedStars={ratings}
              starHeight={15}
            />
          </View>
          {/* <Text style={styles.reviews}>{reviewsCount} Reviews</Text> */}
        </View>
        <MaterialCommIcons
          name="chevron-right"
          size={40}
          color={Colors.black}
          style={styles.chevron}
        />
      </View>
    </TouchableOpacity>
  )
}

export default Restaurant
