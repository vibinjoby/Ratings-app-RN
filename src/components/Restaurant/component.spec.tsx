import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import Restaurant from './component'

describe('Restaurant', () => {
  it('should render', () => {
    const { toJSON } = render(
      <Restaurant
        restaurantName="California Roll"
        ratings={4.5}
        reviewsCount={15}
        onPress={() => console.log('pressed')}
      />,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
