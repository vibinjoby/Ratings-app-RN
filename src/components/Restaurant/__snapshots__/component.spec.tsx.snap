// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant should render 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://source.unsplash.com/user/picoftasty",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "height": 80,
          "marginEnd": 20,
          "width": 80,
        }
      }
    />
    <View>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#181725",
            "fontFamily": "Poppins-Regular",
            "fontSize": 24,
            "lineHeight": 29,
            "marginTop": 5,
            "maxWidth": 200,
          }
        }
      >
        California Roll
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#7C7C7C",
              "fontFamily": "Poppins-Regular",
              "fontSize": 12,
              "marginRight": 10,
            }
          }
        >
          4.5
        </Text>
        <RCTScrollView
          ItemSeparatorComponent={[Function]}
          data={
            Array [
              1,
              2,
              3,
              4,
              5,
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={true}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEnabled={false}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#FF5823",
                        "fontSize": 15,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="font-icon"
                >
                  
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "marginRight": 5,
                    },
                    undefined,
                  ]
                }
              />
            </View>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#FF5823",
                        "fontSize": 15,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="font-icon"
                >
                  
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "marginRight": 5,
                    },
                    undefined,
                  ]
                }
              />
            </View>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#FF5823",
                        "fontSize": 15,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="font-icon"
                >
                  
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "marginRight": 5,
                    },
                    undefined,
                  ]
                }
              />
            </View>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#FF5823",
                        "fontSize": 15,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="font-icon"
                >
                  
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "marginRight": 5,
                    },
                    undefined,
                  ]
                }
              />
            </View>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#FF5823",
                        "fontSize": 15,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="font-icon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#181725",
            "fontSize": 40,
          },
          Object {
            "position": "absolute",
            "right": 10,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰅂
    </Text>
  </View>
</View>
`;
